app/controllers/compute_resources_controller.rb

```diff
@@ -94,6 +94,7 @@ def test_connection
       @compute_resource = ComputeResource.new_provider(params[:compute_resource])
     end
     @compute_resource.test_connection :force => true
+    @success = true if @compute_resource.errors[:base].blank?
     render :partial => "compute_resources/form", :locals => { :compute_resource => @compute_resource }
```

Comment:
How about making success a local variable and passing it in the `:locals` hash? [Some background](http://thepugautomatic.com/2013/05/locals)